lib/propolis/src/block/crucible.rs

```diff
@@ -29,11 +82,26 @@ impl CrucibleBackend {
         request: VolumeConstructionRequest,
         read_only: bool,
         producer_registry: Option<ProducerRegistry>,
+        my_address: SocketAddr,
+        log: slog::Logger,
     ) -> io::Result<Arc<Self>> {
+        // If provided an address for propolis server, figure out if it
+        // is IPv6 and if so, store that address for possible later use.
+        let address = match my_address {
+            SocketAddr::V6(my_address) => Some(*my_address.ip()),
+            SocketAddr::V4(_) => None,
+        };
```

Comment:
I'd be inclined to phrase this a little more explicitly as "should this backend try to contact Nexus after scrubbing completes, and if so, how should it obtain Nexus's address?" It seems legitimate to me to have a Propolis server that's serving with an IPv6 address but that's not part of an Omicron deployment (e.g. we might want to do this in CI someday to enable multi-machine tests with links fast enough not to limit live migration speeds). This is probably harmless in this change (it'll just produce spurious errors in `remove_read_only_parent`), but I think it's worthwhile to have this path be explicit about what it's configuring.

The exact method by which I'd do this probably depends on whether the pantry also has to notify Nexus when scrubbing attempts complete. If it does, then maybe the "scrub notification config" information should go into the VCR so that the `Volume` can do the notifying directly. If the pantry is going to do something else, or there are other reasons not to put this in the VCR, we could instead put this configuration into the `Crucible` variant of `InstanceSpec::StorageBackendKind` (and then have sled agent make sure to specify it while other instance spec clients like PHD leave it blank).

Reply:
I can update the comment to better reflect your wording above, as this is indeed for Nexus notification.

The pantry may/will have to support a scrub, but it depends on how long pantry instances are going to be around.

Another piece to consider here is that crucible or even other parts of propolis may want to send other messages/errors/warnings to Nexus, and how might we fit those into this framework as well?  If we include
the Nexus contact info in the VCR, then Propolis might not be able to use it.

Reply:
> Another piece to consider here is that crucible or even other parts of propolis may want to send other messages/errors/warnings to Nexus, and how might we fit those into this framework as well? If we include
the Nexus contact info in the VCR, then Propolis might not be able to use it.

In this case, I'd be inclined to treat the Nexus client like the Oximeter producer registry (server gets static configuration parameters that define whether and how to construct a Nexus client; during ensure, create a client if needed, then pass it to the VM controller to hand down to any objects that are interested in it). To be honest, I kind of like the idea of doing that anyway, just to keep all of the "external services that individual entities might or might not care to talk to" abstractions in the same place.

IMO that approach would also make it more palatable to have a `--create-nexus-client` command line option that more clearly describes whether the server's creator intends for the server to be able to talk to Nexus, irrespective of the server address the creator chose.

Reply:
So, currently, inside `crucible.rs` we do this:
```
    async fn _create(
        request: VolumeConstructionRequest,
        read_only: bool,
        producer_registry: Option<ProducerRegistry>,
        my_address: Option<Ipv6Addr>,
        log: slog::Logger,
    ) -> Result<Arc<Self>, crucible::CrucibleError> {
```
The idea would be to do this step, the translation to the actual nexus client, earlier in propolis, then just pass the resulting client info in with the  `create()`?

Reply:
I've updated the PR, the latest changes should have this implemented as (I thought) you suggested.